### PR TITLE
remove locale from url

### PR DESCRIPTION
**Pull request by LBruegger**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -74,7 +74,7 @@ def prepare_menu
   end
 
   def default_url_options(options = {})
-    { locale: I18n.locale }.merge options
+    { locale: nil }.merge options
```
Comment (TheWalkingLeek) [inline app/controllers/application_controller.rb]: This is useless now, isn't it?
If you set the locale to nil, it's the same as not setting it at all. Merging it with options just returns options

File: config/routes.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@
   get 'status/health', to: 'status#health'
   get 'status/readiness', to: 'status#readiness'
 
-  scope "(:locale)", locale: /en|de|fr|zh|ru|ch_be/ do
+  scope "/" do
```
Comment (TheWalkingLeek) [inline config/routes.rb]: Is this even needed now?
Can't you just remove the scope completely? Feels like that should have the same effect

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (LBruegger) [reply app/controllers/application_controller.rb]: Sure. I didn't know how relevant this, in our case is.

File: config/routes.rb (hunk 2 above)
Comment (LBruegger) [reply config/routes.rb]: Yes, you're right. It is completely unnecessary.

**Comment by psunix** [comment]
pretty good solution ... and it works :)

some room for improvements:

- create a file in config/routes/legacy.rb (we're going to have some more legacy routes in the future) and add the one-liner from the routes.rb
- rename DefaultRoutingController -> LegacyRoutesController
- create dir app/controllers/legacy_routes
- move RemoveLocale to app/controllers/legacy_routes/locale.rb
- nest Locale inside legacy_routes namespace -> LegacyRoutes::Locale
- create integration test file: test/integration/legacy_routes/locale_test.rb and make sure redirecting is working as expected
- check other comments

File: app/controllers/default_routing_controller.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2008-2017, Puzzle ITC GmbH. This file is part of
+#  Cryptopus and licensed under the Affero General Public License version 3 or later.
+#  See the COPYING file at the top-level directory or at
+#  https://github.com/puzzle/cryptopus.
+
+class DefaultRoutingController
+
+  def redirect
```
Comment (psunix) [inline app/controllers/default_routing_controller.rb]: I think it's completely save to just skip authorization here completely. 
make sure you're testing all cases in the new integration test.
logged in and not logged in user for example

Comment (psunix) [reply app/controllers/default_routing_controller.rb]: if the user is not logged in, he's going to be redirected to the login form. we also remember what URL the user tried to access and redirect him after successful login. 
check if this is also possible with the legacy url

File: app/policies/default_policy.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
```
Comment (psunix) [inline app/policies/default_policy.rb]: remove policy !

File: app/utils/remove_locale.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2008-2017, Puzzle ITC GmbH. This file is part of
+#  Cryptopus and licensed under the Affero General Public License version 3 or later.
+#  See the COPYING file at the top-level directory or at
+#  https://github.com/puzzle/cryptopus.
+
+class RemoveLocale
+
+  LOCALES = I18n.available_locales.freeze
```
Comment (psunix) [inline app/utils/remove_locale.rb]: LOCALES_REGEX = I18n.available_locales.map do |locale| ... .freeze

File: app/utils/remove_locale.rb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+  def initialize
+    @regex_pattern = build_pattern
+  end
+
+  def prepare_url(url)
```
Comment (psunix) [inline app/utils/remove_locale.rb]: remove_locale(url)

just return new url without locale in it

File: app/utils/remove_locale.rb [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    prepared_url
+  end
+
+  def changed?
+    @changed
+  end
+
+  private
+
+  def not_equal(url1, url2)
```
Comment (psunix) [inline app/utils/remove_locale.rb]: =! already means not_equal :)
always add ? to boolean functions

File: test/integration/legacy_routes/locale_test.rb [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+  test 'redirects to account url without locale' do
+    login_as('bob')
+    accounts_url = team_group_accounts_path(teams(:team1).id, groups(:group1).id)
+    get "/de/teams/#{teams(:team1).id}/groups/#{groups(:group1).id}/accounts"
+
+    assert_redirected_to accounts_url
+  end
+
+  test 'redirects to profile url with locale' do
```
Comment (psunix) [inline test/integration/legacy_routes/locale_test.rb]: without locale !

**Pull request merged**